packages/sdk-v2/src/sdk/claim/utils/getRecentClaims.ts

```diff
@@ -15,80 +17,105 @@ export const getRecentClaims = async (
+      const apiUrl = `${baseURL}${queryString}`;
+
+      const result = await fetch(apiUrl).then(res => res.json());
+      if (result?.status !== "1") {
+        throw new Error("Failed to fetch events from explorer");
       }
+      return result.result;
     });
+    const fallbackResult = await fallback(apiCalls);
+    return fallbackResult;
+  };
 
-    if (callResults.length > 0) {
-      const transactionsSorted = orderBy(callResults, ["timeStamp"], ["desc"]);
-
-      const formatted = transactionsSorted.map((tx: any) => {
-        const { address, data, timeStamp, transactionHash } = tx;
-
-        const isGdPool = tx.topics[0] === claimHash;
-
-        const claimAmount = ethers.BigNumber.from(data);
-        const date = moment(ethers.BigNumber.from(timeStamp).toNumber() * 1000).utc();
-
-        // todo: add utilty getContractName(address) to read pool name from registry
-        // for now we assume its either GoodDollar or RedTent
-        const contractName = isGdPool ? "GoodDollar" : "RedTent";
+  const fetchLogsFromBlockchain = async (hash: string, pool: string) => {
+    const result = await library.getLogs({
+      address: pool,
+      topics: [hash, sender32],
+      fromBlock: "0x" + blockNoStart.toString(16),
+      toBlock: "0x" + lastBlockNo.toString(16)
+    });
 
-        return { address, claimAmount, contractName, date, transactionHash, isPool: !isGdPool };
-      });
+    if (!isArray(result)) throw new Error("Failed to fetch events from rpc");
 
-      return formatted;
-    }
+    const withTimestamp = await Promise.all(
+      result.map(async log => {
+        const b = await library.getBlock(log.blockNumber);
+        log.timeStamp = b.timestamp;
+        return log;
+      })
+    );
+    return withTimestamp;
+  };
+  try {
+    const results = await Promise.all(
+      flatten(
+        pools.map(pool =>
+          hashes.map(hash =>
+            fetchLogs(hash, pool).catch(() => {
+              return fetchLogsFromBlockchain(hash, pool);
+            })
+          )
+        )
+      )
+    );
+    const callResults: any[] = flatten(results).filter(_ => _);
+
+    const transactionsSorted = orderBy(callResults, ["blockNumber"], ["desc"]);
+
+    const formatted = transactionsSorted.map((tx: any) => {
+      const { address, data, timeStamp, transactionHash } = tx;
+
+      const isGdPool = tx.topics[0] === claimHash;
+
+      const claimAmount = ethers.BigNumber.from(data);
+      const date = moment(ethers.BigNumber.from(timeStamp).toNumber() * 1000).utc();
+
+      // todo: add utilty getContractName(address) to read pool name from registry
+      // for now we assume its either GoodDollar or RedTent
+      const contractName = isGdPool ? "GoodDollar" : "RedTent";
+
+      return { address, claimAmount, contractName, date, transactionHash, isPool: !isGdPool };
+    });
 
-    return callResults;
+    const merged = formatted.concat(claimsCache);
```

Comment:
@sirpy 
there is an issue with using claimsCache from local-storage.
we expect on UI `date` to be moment object.
because we now pull in the claims-cache, moment is serialized and is just a string of date.

so either here we need to convert them to moment objects.
or UI flow has to change to handle date-strings instead of moment objects.

your suggestion?

Reply:
date as string handled here now: https://github.com/GoodDollar/GoodWeb3-Mono/pull/216/commits/f191836b9ea0671fca8d37885c497a256fcbffb8